lib/types/db.ts

```diff
@@ -0,0 +1,53 @@
+import Sequelize, { DataTypeAbstract, DefineAttributeColumnOptions } from 'sequelize';
+import { db } from './index';
+
+export type SequelizeAttributes<T extends { [key: string]: any }> = {
+  [P in keyof T]: string | DataTypeAbstract | DefineAttributeColumnOptions
+};
+
+export type CurrencyFactory = {
+  id: string;
+};
+
+export type CurrencyAttributes = CurrencyFactory;
```

Comment:
How should `Instance` differ from `Attributes`? In my mind it would make sense for `Attributes` to be the metadata that `xud` adds to particular objects, and that `Instance` to be a combination of the properties from the `Attributes` type and `Factory` (or "`Request`") type, but maybe I'm not following completely. It might also be helpful to throw in a few comments here since this was not easy for me to grasp right away.

Reply:
I answered in other comments, but here's the differences:

Factory - what you need to provide to create a new record
Attributes - the full definition of the record
Instance - Attributes + db operations properties (you can look at `@types/sequelize`)